innovation_sweet_spots/tests/data_update_checks/utils.py

```diff
@@ -0,0 +1,193 @@
+    else:
+        logger.warning(f"{start_message} the same number of {end_message}.")
+
+
+def check_cols_match(
+    cur_df: pd.DataFrame,
+    new_df: pd.DataFrame,
+    filename: str,
+    cur_dir: str,
+    new_dir: str,
+):
+    """Check that the number of columns in the current and new dataframes are
+    the ame. Log the outcome and provide additional information on the
+    differences."""
+    cur_cols = set(cur_df.columns)
+    new_cols = set(new_df.columns)
+    if cur_cols == new_cols:
+        logger.info(f"{filename} in {cur_dir}/ and {new_dir}/ columns match.")
+    else:
+        logger.warning(f"{filename} in {cur_dir}/ and {new_dir}/ columns do not match.")
+        if missing_cols := cur_cols - new_cols:
+            logger.warning(
+                f"Compared with {cur_dir}/{filename}, {new_dir}/{filename} dataset is missing {len(missing_cols)} columns: {missing_cols}."
+            )
+        if add_cols := new_cols - cur_cols:
+            logger.warning(
+                f"Compared with {cur_dir}/{filename}, {new_dir}/{filename} dataset has {len(add_cols)} additional columns: {add_cols}."
+            )
+
+
+def calc_percent_nans(df: pd.DataFrame) -> pd.DataFrame:
+    """Return percentage of nans in each column for provided dataframe."""
+    return (
+        (df.isnull().sum() / len(df) * 100)
+        .reset_index()
+        .rename(columns={"index": "column", 0: "% nans"})
+    )
+
+
+def combine_data_for_nan_plot(
+    cur_df: pd.DataFrame, new_df: pd.DataFrame, cur_dir: str, new_dir: str
+) -> pd.DataFrame:
+    """Combine new and current dataframes and add a version column."""
+    return pd.concat(
+        [
+            calc_percent_nans(new_df).assign(version=new_dir),
+            calc_percent_nans(cur_df).assign(version=cur_dir),
+        ]
+    )
+
+
+def nan_plot(
+    combined_data: pd.DataFrame, filename: str, cur_dir: str, new_dir: str, dataset: str
+):
+    """Plot nan percentage for each column, comparing the new
+    and current data
+
+    Args:
+        combined_data: Data to be used to make the plot
+        filename: File to be checked
+        cur_dir: Current version data directory
+        new_dir: New version data directory
+        dataset: "cb", "gtr" or "nihr"
+    """
+    nan_plot = sns.catplot(
+        data=combined_data,
+        kind="bar",
+        x="% nans",
+        y="column",
+        hue="version",
+        orient="h",
+        height=7,
+        palette="Paired",
+    ).set(
+        title=f"Percentage of NaNs in {cur_dir} vs. {new_dir} for {filename}",
+        xlim=(0, 1),
+    )
+    save_lbl = filename.split(".")[0]
+    save_dir = DUC_DIR / f"{dataset}/{cur_dir}_vs_{new_dir}_plots/"
+    if not save_dir.exists():
+        save_dir.mkdir(parents=True)
+    save_path = save_dir / f"{save_lbl}_nan_plot.png"
+    nan_plot.figure.savefig(save_path, bbox_inches="tight")
+    logger.info(f"NaN plot saved to {save_path}")
+
+
+def run_data_checks(
```

Comment:
Hey @Jack-0-0 , would it be possible to slightly generalise this function by allowing a case where the `dataset = None`, and the util could still compare the tables between both folders, but wouldn't run the dataset-specific tests?